File: Mobile/Throw/src/uk/co/robhemsley/flow/trans/Msg.java
package uk.co.robhemsley.flow.trans;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.Random;

import org.codehaus.jackson.JsonGenerationException;
import org.codehaus.jackson.annotate.JsonIgnore;
import org.codehaus.jackson.annotate.JsonProperty;
import org.codehaus.jackson.map.JsonMappingException;
import org.codehaus.jackson.map.ObjectMapper;

public class Msg {
	@JsonIgnore
	public static int TYPE_SYSTEM = 0;
	@JsonIgnore
	public static int TYPE_DATA = 1;
	@JsonIgnore
	public static int TYPE_DEBUG = 2;
	@JsonIgnore
	public static int TYPE_CONN_DETAILS = 3;
	@JsonIgnore
	public static int TYPE_FILE_INFO = 4;
	@JsonIgnore
	public static int TYPE_FILE_CHUNK = 5;
	@JsonIgnore
	public static int TYPE_ACK = 6;
	@JsonIgnore
	public static int TYPE_FILE_REQUEST = 7;
	@JsonIgnore
	public static int TYPE_FILE_PUSH = 8;
	
	private int type;
	private String body;
	private long timestamp;
	private String msg_id;
	
	public Msg(String usr_id){
		this.setMsgId(usr_id);
		this.setTimestamp(System.currentTimeMillis()/1000);
		
		Random randomGenerator = new Random();
		this.setMsgId(usr_id+"-"+new Integer(randomGenerator.nextInt(9999999)).toString());
	}

	@JsonProperty("type")
	public int getType() {
		return type;
	}

	public void setType(int type) {
		this.type = type;
	}

	@JsonProperty("body")
	public String getBody() {
		return body;
	}

	public void setBody(String body) {
		this.body = body;
	}

	@JsonProperty("timestamp")
	public long getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(long timestamp) {
		this.timestamp = timestamp;
	}

	@JsonProperty("msg_id")
	public String getMsgId() {
		return msg_id;
	}

	public void setMsgId(String msg_id) {
		this.msg_id = msg_id;
	}
	
	@JsonIgnore
	public String getAsJSON() {
	    ObjectMapper mapper = new ObjectMapper();
	    try {
			return mapper.writeValueAsString(this) ;
		} catch (JsonGenerationException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (JsonMappingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} 
	    return "";
	}

}

File: Server/ThrowServer/Msg.py
'''
Created on Jan 24, 2013

@author: rob
'''
import time, json, random

class MsgCon:
    """
    MsgCon - Class
        Provides constant variables for use with the message object.
    """
    TYPE_SYSTEM = 0
    TYPE_DATA = 1
    TYPE_DEBUG = 2
    TYPE_CONN_DETAILS = 3
    TYPE_FILE_INFO = 4;
    TYPE_FILE_CHUNK = 5;
    TYPE_ACK = 6;
    TYPE_FILE_REQUEST = 7;
    TYPE_FILE_PUSH = 8;


class Msg:
    _data = None
    
    def __init__(self, data = None, usr_id = None):
        if usr_id == None:
            if data != None:
                usr_id = data["msg_id"].split("-")[0]
            
        if data == None:
            self._data = {"type": "", "body": "", "timestamp": "", "msg_id": "%s-%d"%(usr_id, random.randint(0,999999))}
        else:
            self._data = data
    
    def get_value(self, key):
        if key in self._data:
            return self._data[key]
        else:
            return None
    
    def set_value(self, key, value):
        self._data[key] = value
        
    def set_type(self, value):
        self.set_value("type", value)
        
    def get_type(self):
        return self.get_value("type")
        
    def set_body(self, value):
        self.set_value("body", value)
        
    def get_body(self):
        return self.get_value("body")

    def set_msg_id(self, value):
        self.set_value("msg_id", value)
        
    def get_msg_id(self):
        return self.get_value("msg_id")
        
    def set_timestamp(self, value):
        self.set_value("timestamp", value)
        
    def get_timestamp(self):
        return self.get_value("timestamp")
    
    def __str__(self):
        self.set_value("timestamp" ,str(time.time()))
        return str(json.dumps(self._data))
    
    def __repr__(self):
        return self.__str__()